music_assistant/providers/roku_media_assistant/player.py

```diff
@@ -0,0 +1,480 @@
+        # This method should handle the play_media command for the given player.
+        # It will be called when media needs to be played on the player.
+        # The media object contains all the details needed to play the item.
+
+        # In 99% of the cases this will be called by the Queue controller to play
+        # a single item from the queue on the player and the uri within the media
+        # object will then contain the URL to play that single queue item.
+
+        # If your player provider does not support enqueuing of items,
+        # the queue controller will simply call this play_media method for
+        # each item in the queue to play them one by one.
+
+        # In order to support true gapless and/or enqueuing, we offer the option of
+        # 'flow_mode' playback. In that case the queue controller will stitch together
+        # all songs in the playbook queue into a single stream and send that to the player.
+        # In that case the URI (and metadata) received here is that of the 'flow mode' stream.
+
+        # Examples of player providers that use flow mode for playback by default are AirPlay,
+        # SnapCast and Fully Kiosk.
+
+        # Examples of player providers that optionally use 'flow mode' are Google Cast and
+        # Home Assistant. They provide a config entry to enable flow mode playback.
+
+        # Examples of player providers that natively support enqueuing of items are Sonos,
+        # Slimproto and Google Cast.
+
+        if not (queue := self.mass.player_queues.get_active_queue(self.player_id)):
+            return
+
+        try:
+            device_info = await self.roku.update()
+
+            app_running = False
+
+            if device_info.app is not None:
+                app_running = (
+                    device_info.app.app_id == self.provider.config.get_value(CONF_ROKU_APP_ID)
+                    if not device_info.app.screensaver
+                    else False
+                )
+
+            current_duration = 0
+
+            if queue.current_item is not None and queue.current_item.media_item is not None:
+                current_duration = cast("int", queue.current_item.media_item.duration)
+
+            f_media = {
+                "u": media.uri,
+                "t": "a",
+                "albumName": media.album,
+                "songName": media.title,
+                "artistName": "Music Assistant Radio"
+                if media.media_type == MediaType.RADIO
+                else media.artist,
+                "albumArt": media.image_url,
+                "songFormat": "flac",
+                "duration": "" if media.duration is None else current_duration,
+                "timeOffset": "" if media.duration is None else (current_duration - media.duration),
+                "isLive": "true" if media.media_type == MediaType.RADIO else "",
+            }
+
+            if queue.flow_mode and queue.current_item:
```

Comment:
In flow mode, you will only receive a single audio stream.
unless you have some way to dynamically overwrite the currently playing media or your device supports ICY metadata, there is no way to get the media details.

This implementation here just sets the media info from the first track playing and then it will be wrong when the next track plays.